src/GrasshopperTeklaDrawingLink/Components/Text/FontAttributesComponent.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections.Generic;
+using System.Drawing;
+
+using Grasshopper.Kernel;
+
+using GTDrawingLink.Properties;
+using GTDrawingLink.Tools;
+using GTDrawingLink.Types;
+
+using Tekla.Structures.Drawing;
+
+
+namespace GTDrawingLink.Components.Text
+{
+    public class FontAttributesComponent : TeklaComponentBase
+    {
+        public override GH_Exposure Exposure => GH_Exposure.primary;
+
+        public FontAttributesComponent()
+            : base(ComponentInfos.FontAttributesComponent)
+        {
+        }
+
+        protected override void RegisterInputParams(GH_InputParamManager pManager)
+        {
+            SetParametersAsOptional(pManager, new List<int> {
+               pManager.AddParameter(new EnumParam<DrawingColors>(ParamInfos.DrawingColor, GH_ParamAccess.item)),
+
+               pManager.AddTextParameter(
```

Comment:
Please use wrappers around GH_InstanceDescription from the TeklaComponentBase class like protected void _AddTextParameter(GH_InputParamManager pManager, GH_InstanceDescription paramInfo, GH_ParamAccess access, bool optional = false)_ .
Their role is to shorten declaration of inputs, to reduce time needed to check if all of the provided Name, NickName, Description belongs to the same GH_InstanceDescription.